app/actions/remote/search.ts

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {SYSTEM_IDENTIFIERS} from '@app/constants/database';
+import {ActionType} from '@constants';
+import DatabaseManager from '@database/manager';
+import NetworkManager from '@init/network_manager';
+import {queryCurrentUser} from '@queries/servers/user';
+
+import {forceLogoutIfNecessary} from './session';
+
+import type {Client} from '@client/rest';
+
+type PostSearchRequest = {
+    error?: unknown;
+    order?: string[];
+    posts?: Post[];
+}
+
+export async function getRecentMentions(serverUrl: string): Promise<PostSearchRequest> {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    let client: Client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch (error) {
+        return {error};
+    }
+
+    let data;
+
+    try {
+        const currentUser = await queryCurrentUser(operator.database);
+        if (!currentUser) {
+            return {
+                posts: [],
+                order: [],
+            };
+        }
+        const terms = currentUser.mentionKeys.map(({key}) => key).join(' ').trim() + ' ';
+        data = await client.searchPosts(terms, true);
+    } catch (error) {
+        forceLogoutIfNecessary(serverUrl, error as ClientErrorProps);
+        return {error};
+    }
+
+    const {posts, order} = await processPostsFetched(serverUrl, ActionType.SEARCH.RECEIVED_MENTIONS, data);
```

Comment:
I couldn't find what `ActionType.SEARCH.RECEIVED_MENTIONS` does, unless is you trying skip postsInChannel & postsInThread, which I guess that is what you are looking for? in that case, using an empty string would also have the same result, right ?